packages/core/src/services/evaluations/run.ts

```diff
@@ -101,52 +101,50 @@ export const runEvaluation = async (
     required: ['result', 'reason'],
   }
 
-  const chainResult = await runChain({
+  const providersMap = await buildProvidersMap({
+    workspaceId: evaluation.workspaceId,
+  })
+  const run = await runChain({
     workspace,
     chain,
     source: LogSources.Evaluation,
-    apikeys: await buildProviderApikeysMap({
-      workspaceId: evaluation.workspaceId,
-    }),
+    providersMap,
     configOverrides: {
       schema,
       output: 'object',
     },
   })
 
-  if (chainResult.error) return chainResult
-
-  // Call the new method to handle the promise
-  chainResult.value.response.then((response) =>
-    handleEvaluationResponse(response, documentUuid, evaluation, documentLog),
+  const response = run.response as Promise<ChainStepResponse<'object'>>
+  response.then((res) =>
+    handleEvaluationResponse(res, documentUuid, evaluation, documentLog),
   )
 
-  return chainResult
+  return Result.ok(run)
```

Comment:
why

Reply:
I removed `Result` from `runChain` but this service was returning Result and don't want to introduce more entropy